### PR TITLE
AC-621:Add Floating Action Button in patient's Details tab

### PR DESCRIPTION
**Pull request by deepak140596**
## Description of what I changed

Added FAB to Coordinator Layout of activity_patient_dashboard.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-621

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/details/PatientDetailsFragment.java [hunk 1]
```diff
@@ -133,6 +130,7 @@ public void resolvePatientDataDisplay(final Patient patient) {
         }
 
         if (null != patient.getAddress()) {
+            ((TextView) rootView.findViewById(R.id.addressDetailsStreet)).setText(patient.getAddress().getAddressString());
```
Comment (deepak140596) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/details/PatientDetailsFragment.java]: This line was missed in the AC-620 that caused the address textview to be empty

File: openmrs-client/src/main/res/layout/fragment_patient_details.xml [hunk 2]
```diff
@@ -24,245 +24,237 @@
         android:layout_width="match_parent"
```
Comment (deepak140596) [inline openmrs-client/src/main/res/layout/fragment_patient_details.xml]: This layout was reformatted using AS.  Also test texts were removed .

File: openmrs-client/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -40,5 +40,12 @@
     <dimen name="label_margin_wide">32dp</dimen>
     <dimen name="big_heading">18dp</dimen>
     <dimen name="small_heading">16dp</dimen>
+    <dimen name="fab_size_mini">40dp</dimen>
+    <dimen name="fab_size_normal">56dp</dimen>
+    <dimen name="fab_bottom_margin">4dp</dimen>
+    <dimen name="standard_55">55dp</dimen>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/dimens.xml]: Name this based on what it is used for.

File: openmrs-client/src/main/res/values/dimens.xml [hunk 4]
```diff
@@ -40,5 +40,12 @@
...
+    <dimen name="standard_105">105dp</dimen>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/dimens.xml]: Name this based on what it is used for.

File: openmrs-client/src/main/res/values/dimens.xml [hunk 5]
```diff
@@ -40,5 +40,12 @@
...
+    <dimen name="standard_155">155dp</dimen>
+    <dimen name="standard_21">21dp</dimen>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/dimens.xml]: Name this based on what it is used for.

File: openmrs-client/src/main/res/values/dimens.xml [hunk 6]
```diff
@@ -40,5 +40,12 @@
     <dimen name="label_margin_wide">32dp</dimen>
     <dimen name="big_heading">18dp</dimen>
     <dimen name="small_heading">16dp</dimen>
+    <dimen name="fab_size_mini">40dp</dimen>
+    <dimen name="fab_size_normal">56dp</dimen>
+    <dimen name="fab_bottom_margin">4dp</dimen>
+    <dimen name="standard_55">55dp</dimen>
+    <dimen name="standard_105">105dp</dimen>
+    <dimen name="standard_155">155dp</dimen>
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/values/dimens.xml]: Name this based on what it is used for.

File: openmrs-client/src/main/res/layout/activity_patient_dashboard.xml [hunk 7]
```diff
@@ -76,4 +76,40 @@
...
+        app:rippleColor="@color/grey" />
+
+    <com.google.android.material.floatingactionbutton.FloatingActionButton
+        android:id="@+id/activity_patient_dashboard_action_fab"
+        android:layout_width="@dimen/fab_size_normal"
+        android:layout_height="wrap_content"
+        android:layout_gravity="end|bottom"
+        android:layout_margin="@dimen/margin_long"
+        android:hapticFeedbackEnabled="true"
+        android:src="@drawable/ic_add"
```
Comment (f4ww4z) [inline openmrs-client/src/main/res/layout/activity_patient_dashboard.xml]: I think that the Pencil icon is more suitable than the + icon.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java [hunk 8]
```diff
@@ -124,4 +139,52 @@ public void setBackdropImage(Bitmap backdropImage, String patientName) {
...
+    }
+
+    public void startPatientUpdateActivity(long patientId) {
+        Intent updatePatient = new Intent(this, AddEditPatientActivity.class);
+        updatePatient.putExtra(ApplicationConstants.BundleKeys.PATIENT_ID_BUNDLE,
+                String.valueOf(patientId));
+        startActivity(updatePatient);
+    }
+
+    @SuppressLint("RestrictedApi")
```
[the remark this answers is not in the document]
Comment (deepak140596) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java]: The warning is:

VisibilityAwareImageButton.setVisibility can only be called from within the same library group (groupId=com.google.android.material) less... (Ctrl+F1) 
Inspection info:This API has been flagged with a restriction that has not been met.

[the remark this answers is not in the document]
Comment (deepak140596) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/patientdashboard/PatientDashboardActivity.java]: I wasn't able to change visibility from other fragments.

**Comment by deepak140596** [comment]
@f4ww4z , updated the code. Please review it.

**Pull request merged**